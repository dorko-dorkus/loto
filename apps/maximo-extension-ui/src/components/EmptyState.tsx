import React from 'react';
import clsx from 'clsx';

interface EmptyStateProps {
  icon: React.ReactNode;
  title: string;
  description?: string;
  action?: React.ReactNode;
  className?: string;
}

export default function EmptyState({
  icon,
  title,
  description,
  action,
  className,
}: EmptyStateProps) {
  return (
    <div
      className={clsx(
        'flex flex-col items-center justify-center gap-4 p-8 text-center',
        className
      )}
    >
      <div className="text-6xl" aria-hidden="true">
        {icon}
      </div>
      <div>
        <h3 className="text-lg font-semibold">{title}</h3>
        {description && (
          <p className="mt-2 text-sm text-gray-600">{description}</p>
        )}
      </div>
      {action && <div>{action}</div>}
    </div>
  );
}
