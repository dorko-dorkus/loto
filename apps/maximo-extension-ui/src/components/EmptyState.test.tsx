import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import EmptyState from './EmptyState';

const Icon = () => <svg data-testid="icon" />;

const Action = () => <button type="button">Action</button>;

test('matches snapshot', () => {
  const { container } = render(
    <EmptyState
      icon={<Icon />}
      title="Nothing here"
      description="There is currently no data to display"
      action={<Action />}
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
