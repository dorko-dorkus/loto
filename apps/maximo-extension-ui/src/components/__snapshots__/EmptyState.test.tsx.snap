// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div
  class="flex flex-col items-center justify-center gap-4 p-8 text-center"
>
  <div
    aria-hidden="true"
    class="text-6xl"
  >
    <svg
      data-testid="icon"
    />
  </div>
  <div>
    <h3
      class="text-lg font-semibold"
    >
      Nothing here
    </h3>
    <p
      class="mt-2 text-sm text-gray-600"
    >
      There is currently no data to display
    </p>
  </div>
  <div>
    <button
      type="button"
    >
      Action
    </button>
  </div>
</div>
`;
